src/test/java/net/snowflake/client/jdbc/ResultSetMultiTimeZoneLatestIT.java

```diff
@@ -256,6 +260,24 @@ private void helperTestUseSessionTimeZone(
     assertEquals(true, rs.getTime("COLD").toString().equals(expectedTime2));
     // Cannot getTime() for Date column (colE)
 
+    // Test special case for timestamp_tz (offset added)
+    // create table with all timestamp types, time, and date
+    statement.execute("create or replace table tabletz (colA timestamp_tz)");
+    prepSt = connection.prepareStatement("insert into tabletz values(?), (?)");
+    prepSt.setString(1, expectedTimestamp + " +0500");
+    prepSt.setString(2, expectedTimestamp2 + " -0200");
+    prepSt.execute();
+
+    rs = statement.executeQuery("select * from tabletz");
+    rs.next();
+    assertEquals(useSessionTimezone, rs.getTimestamp("COLA").toString().equals(expectedTimestamp));
```

Comment:
Can you add comments to explain what is the expected behavior here? Do you mean if JDBC uses session time zone, then we skip the timezone in timestamp_tz?

Reply:
Added comments! By default, the driver will display the timestamp/time/date values with the JVM offset. So if the JVM is 3 hours ahead of the session timezone, the values will be displayed as -3 hours as they would in the UI. With the session parameter JDBC_USE_SESSION_TIMEZONE, the values are displayed using the session timezone, not the JVM timezone, so there's no offset between the inserted value and the displayed value. With this change, a value inserted like this:
`2019-01-01 17:17:17.6 +0500`
will get displayed like this: 
ResultSet.getTimestamp(): `2019-01-01 17:17:17.6`
ResultSet.getTime(): `17:17:17.6`
ResultSet.getDate(): `2019-01-01`

Reply:
I see. Maybe it is worth to have those comment into the code too.

Reply:
Added comments at the beginning of the test!